content/sensu-go/6.0/reference/filters.md

```diff
@@ -832,6 +832,96 @@ spec:
 
 {{< /language-toggle >}}
 
+## Use built-in JavaScript filter execution functions
+
+Sensu includes four built-in JavaScript functions for event filter execution:
+
+- `sensu.CheckDependencies`
+- `sensu.FetchEvent`
+- `sensu.CheckStatus`
+- `sensu.ListEvents`
+
+You can use these functions to query your event stores.
+
+### `sensu.CheckDependencies`
+
+The `sensu.CheckDependencies` function takes zero or more checks as arguments.
+It returns `true` if all the specified checks are passing or `false` if any of the specified checks are failing.
+
+If you do not specify any checks, the function always returns `true`.
+
+You can refer to the checks as strings:
+
+{{< code javascript >}}
+sensu.CheckDependencies("database", "disk")
+{{< /code >}}
+
+If you pass the check names as strings, Sensu assumes that the entites are the same as those in the events being filtered.
+
+You can also refer to the checks in objects that include both the entity and check name.
+For example:
+
+{{< code javascript >}}
+sensu.CheckDependencies({entity: "server01", check: "disk"}, {entity: "server01", check: "database"})
+{{< /code >}}
+
+In both cases, if no event matches the specified entities and checks, Sensu will raise an error.
+
+### `sensu.EventStatus`
+
+The `sensu.EventStatus` function takes zero or more checks as arguments.
+It returns an array of status codes for the events associated with the specified checks.
+
+You can refer to the checks as strings:
+
+{{< code javascript >}}
+sensu.EventStatus("database", "disk")
+{{< /code >}}
+
+If you pass the check names as strings, Sensu assumes that the entites are the same as those in the events being filtered.
+
+You can also refer to the checks in objects that include both the entity and check name.
+For example:
+
+{{< code javascript >}}
+sensu.EventStatus({entity: "server01", check: "disk"}, {entity: "server01", check: "database"})
+{{< /code >}}
+
+In both cases, if no event matches the specified entities and checks, Sensu will raise an error.
+
+### `sensu.FetchEvent`
+
+The `sensu.FetchEvent` function loads the Sensu events that correspond to the specified entity and check names.
+
+The format is `sensu.FetchEvent(entity, check)`.
+For example:
+
+{{< code javascript >}}
+sensu.FetchEvent("server01", "disk")
+{{< /code >}}
+
+You can only load events from the same namespace as the event being queried.
```

Comment:
Something seems not quite right with this statement. Should it be "the entity being queried" instead? As written, it seems circular maybe (...events from...the event being queried).

Reply:
That should say "the event being filtered"

Reply:
ohhh right I see heh. I was trying to avoid using "filtered" in a general sense but we are actually talking about the subject event for the event filter. I get it now, thank you.